crates/lunatic-common-api/src/lib.rs

```diff
@@ -11,6 +11,37 @@ pub fn get_memory<T>(caller: &mut Caller<T>) -> std::result::Result<Memory, Trap
         .or_trap("Export `memory` is not a memory")
 }
 
+// Call guest to allocate a Vec of size `size`
+pub fn allocate_guest_memory<'a, T: Send>(
+    caller: &'a mut Caller<T>,
+    size: u32,
+) -> Pin<Box<dyn Future<Output = Result<u32, Trap>> + Send + 'a>> {
+    Box::pin(async move {
+        let mut results = [Val::I32(0)];
+        let result = caller
+            .get_export("alloc")
```

Comment:
This is probably not safe for AssemblyScript users. This would also probably require the runtime to be exported, which is feasable, but I'm not sure what a good solution is here.

Reply:
@jtenner Would manually exporting a function `alloc` that just calls `malloc` (or the AS equivalent) work? If you can allocate memory manually there is no reason why you couldn't just wrap it into an exported function with a well defined name.

Reply:
@jtenner I added the ability to set a different "allocator" function (called `set_custom_guest_allocator`) which will allow you set to set custom allocators for each connection. All you'd need to do is provide the allocator function and set it every time when creating a new sqlite connection.